src/Nethermind/Nethermind.Synchronization/ParallelSync/SyncDispatcher.cs

```diff
@@ -137,6 +149,14 @@ public async Task Start(CancellationToken cancellationToken)
                 if (Logger.IsWarn) Logger.Warn($"Failure when executing request {e}");
             }
 
+            if (Feed.IsMultiFeed)
+            {
+                // Don't free if num of concurrent processing threads is limited
+                // Otherwise, it'll keep downloading more requests.
+                await _concurrentProcessingSemaphore.WaitAsync(cancellationToken);
+                Free(allocation);
+            }
+
```

Comment:
I don't get it can you make comment easier to understand?

Reply:
How about now?